joomla/joomla-cms
# Fix custom field calendar value when changing category

Pull Request for Issue #30343.

### Summary of Changes
In short: When using a custom field of type calendar in an article and changing the category of said article, the time is changed of the custom field.

In long: When a site/user has a different timezone than UTC and has a custom field of type calendar for his articles and then, when editing such an article, changes its category, the page reloads and processes the data on that reload. In theory the filtered data should be stored to the session and we have according code in FormController around line 950. However that code only checks the first level of the $data array. Custom fields however are in a subarray, which lets this fail.
IMPORTANT: THIS ONLY FIXES THIS FOR THE FIRST LEVEL OF THE SUBARRAY! That means, when you have a form that has a fields tag inside a fields tag inside a fields tag, this issue still exists. I don't know yet how to solve this properly! Help is welcome here.

This is the fix for 3.10, but this issue is most likely also present in 4.0! For 4.0 I would actually expect the field to handle this directly, since right now this is hardcoded to the core Calendar field and thus means third party calendar fields or other fields with similar data handling have to become a lot more creative. 

### Testing Instructions
1. Set the timezone of your site to something else than UTC.
2. Create a custom field for your article of type "Calendar".
3. Have at least 2 categories.
4. Edit an article and set the calendar custom field to a known date and time.
5. change the category and see the page reload on the change.

### Actual result BEFORE applying this Pull Request
The time (and maybe even the date) changes by the offset to UTC.

### Expected result AFTER applying this Pull Request
The time and date stay the same.
